File: src/pattern/abstractFactory/listFactory/java/ListTray.java
package pattern.abstractFactory.listFactory.java;

import pattern.abstractFactory.factory.java.Item;
import pattern.abstractFactory.factory.java.Tray;

public class ListTray extends Tray {
    public ListTray(String caption){
        super(caption);
    }

    @Override
    public String makeHTML() {
        StringBuffer buffer = new StringBuffer();

        buffer.append("<li>\n");
        buffer.append(this.caption+"\n");
        buffer.append("<ul>\n");
        for (Item it :
                this.tray) {
            buffer.append(it.makeHTML());
        }

        buffer.append("</ul>\n");

        return buffer.toString();
    }


}

File: src/pattern/abstractFactory/listFactory/kotlin/ListTray.kt
package pattern.abstractFactory.listFactory.kotlin

import pattern.abstractFactory.factory.kotlin.Tray

class ListTray(caption: String): Tray(caption) {
    override fun makeHTML(): String{

        val buffer = StringBuffer()

        buffer.append("<li>\n")
        buffer.append(this.caption + "\n")
        buffer.append("<ul>\n")
        for (it in this.tray) {
            buffer.append(it.makeHTML())
        }

        buffer.append("</ul>\n")

        return buffer.toString()

    }
}